tests/eosio.msig_tests.cpp

```diff
@@ -1366,4 +1366,170 @@ BOOST_FIXTURE_TEST_CASE( unnecessary_approve_logic, eosio_msig_tester ) try {
+   BOOST_REQUIRE_EQUAL( fc::unsigned_int{0}, trx_trace->action_traces.at(0).closest_unnotified_ancestor_action_ordinal );
+   BOOST_REQUIRE_EQUAL( N(sendinline), action_name{trx_trace->action_traces.at(0).receiver} );
+   BOOST_REQUIRE_EQUAL( N(sendinline), name{trx_trace->action_traces.at(0).act.account} );
+   BOOST_REQUIRE_EQUAL( N(send), name{trx_trace->action_traces.at(0).act.name} );
+   BOOST_REQUIRE_EQUAL( N(bob), name{trx_trace->action_traces.at(0).act.authorization[0].actor} );
+   BOOST_REQUIRE_EQUAL( N(active), name{trx_trace->action_traces.at(0).act.authorization[0].permission} );
+   
+   BOOST_REQUIRE_EQUAL( fc::unsigned_int{2}, trx_trace->action_traces.at(1).action_ordinal );
+   BOOST_REQUIRE_EQUAL( fc::unsigned_int{1}, trx_trace->action_traces.at(1).creator_action_ordinal );
+   BOOST_REQUIRE_EQUAL( fc::unsigned_int{1}, trx_trace->action_traces.at(1).closest_unnotified_ancestor_action_ordinal );
+   BOOST_REQUIRE_EQUAL( N(eosio), action_name{trx_trace->action_traces.at(1).receiver} );
+   BOOST_REQUIRE_EQUAL( N(eosio), name{trx_trace->action_traces.at(1).act.account} );
+   BOOST_REQUIRE_EQUAL( N(reqauth), name{trx_trace->action_traces.at(1).act.name} );
+   BOOST_REQUIRE_EQUAL( N(alice), name{trx_trace->action_traces.at(1).act.authorization[0].actor} );
+   BOOST_REQUIRE_EQUAL( N(perm), name{trx_trace->action_traces.at(1).act.authorization[0].permission} );
+
+   act = get_action(N(eosio.msig), N(approve), {}, mvo()
+                     ("proposer", N(bob))
+                     ("proposal_name", N(first))
+                     ("level", permission_level{N(sendinline), N(eosio.code)})
+   );
+   
+   static transaction trx = reqauth( N(alice), {permission_level{N(alice), N(perm)}}, abi_serializer_max_time );
+   
+   base_tester::push_action( N(eosio.msig), N(propose), N(bob), mvo()
+                              ("proposer", "bob")
+                              ("proposal_name", "first")
+                              ("trx", trx)
+                              ("requested", std::vector<permission_level>{{ N(sendinline), N(eosio.code) }})
+   );
+   BOOST_REQUIRE_EQUAL( false, get_earliest_exec_time(N(bob), N(first)).valid() );
+   produce_blocks();
+   
+   // `approve` shall fail when being sent from the wrong contract
+   BOOST_REQUIRE_EXCEPTION( base_tester::push_action( N(wrongcon), N(send), N(bob), mvo()
+                              ("contract", N(eosio.msig))
+                              ("action_name", "approve")
+                              ("auths", std::vector<permission_level>{})
+                              ("payload", act.data)
+                            ),
+                            eosio_assert_message_exception,
+                            eosio_assert_message_is("wrong contract sent `approve` action for eosio.code permmission")
+   );
+   
+   // `approve` shall succeed when being sent from the wrong contract
```

Comment:
You meant to say "... from the right contract".

Reply:
Corrected comment.